srsue/src/phy/sync_sa.cc

```diff
@@ -364,9 +369,18 @@ void sync_sa::worker_end(const srsran::phy_common_interface::worker_context_t& w
   srsran::rf_timestamp_t tx_time = w_ctx.tx_time; // get transmit time from the last worker
   // todo: tx_time.sub((double)ta.get_sec());
 
+  cf_t* buffer_out = srsran_vec_cf_malloc(tx_buffer.get_nof_samples());
+
   // Check if any worker had a transmission
   if (tx_enable) {
     // Actual baseband transmission
+
+    // added tuning here
+    if (tuner) {
+      srsran_channel_tuner_execute(tuner.get(), tx_buffer.get(0), buffer_out, tx_buffer.get_nof_samples());
+      srsran_vec_cf_copy(buffer_out, tx_buffer.get(0), tx_buffer.get_nof_samples());
```

Comment:
The code here might be incomplete: look at `channel.cc` line 198-247, they iterate through each channel, you might want to do it here. Also check what they did with `out[i]`, `in[i]`, `buffer_out`, the variables seem to be more complicated than you think